### DISCUSSION TITLE
Unexpected Anon columns

### DISCUSSION THREAD

**Discussion by FyZzyss** [sqlalchemy/sqlalchemy#8957] [Usage Questions] [answered]
Hello!
I have the following code:
```
u = User.alias("u")
q = sa.select(
        u.c.id,
        u.c.phone,
        u.c.card,
        u.c.email,
        u.c.user_name,
        u.c.user_last_name,
        u.c.account,
)
```
It is expected that there will be named columns, but I get
` (ImmutableColumnCollection(anon_1.id , anon_1.phone, anon_1.card, anon_1.email, anon_1.user_name, anon_1.user_last_name, anon_1.account). `
How to fix it?
My version is 1.4.41

**Comment by zzzeek** [comment, marked as answer]
hi -

I dont know what you are asking.    the ImmutableColumnCollection shown I would assume is the `q.selected_columns` collection and it shows a series of named columns.

**Reply by FyZzyss** [reply]
Thank you) Some legacy code do hacks to order fields by names with `q.columns` and it works on 1.3.33, but on 1.4.41 fails. With `q.selected_columns` everything works, you save my time!
